shell/server/admin/system-status-server.js

```diff
@@ -1,9 +1,73 @@
-const Crypto = Npm.require("crypto");
+import { Meteor } from "meteor/meteor";
+import { Router } from "meteor/iron:router";
+import { Random } from "meteor/random";
+import Crypto from "crypto";
+import Fs from "fs";
 
 const hashSessionId = function (sessionId) {
   return Crypto.createHash("sha256").update(sessionId).digest("base64");
 };
 
+// An in-memory map of random id -> timestamp, used to authorize requests to download the system
+// log.
+const SYSTEM_LOG_DOWNLOAD_TOKENS = {};
+const SYSTEM_LOG_DOWNLOAD_TOKEN_VALIDITY_DURATION = 60000; // 60 seconds
+
+Meteor.methods({
+  adminGetServerLogDownloadToken() {
+    const db = this.connection.sandstormDb;
+    if (!db.isAdminById(this.userId)) {
+      throw new Meteor.Error(403, "User must be admin to download system log.");
+    }
+
+    const token = Random.id();
```

Comment:
Hmm, technically we ought to be using Random.secret() for these or passing a parameter to Random.id() to make it generate a longer ID.